File: lazarus/scripts/entities/camera.js
import { UserInputs, GameKeys } from '../user_inputs.js'

export class Camera{
    constructor(x, y, scale, target){
	this.x = x
	this.y = y
	this.scale = scale
	this.target = target
    }
    update(dt){
	
	if(UserInputs.pressed_key(GameKeys.ZOOM_IN)){
	    this.scale*=0.99
	} else if(UserInputs.pressed_key(GameKeys.ZOOM_OUT)){
	    this.scale*=1.01
	}
	var epsilon = 0.01
	if(this.scale<this.target.size/20-epsilon){
	    this.scale+=epsilon
	}
	else if (this.scale>this.target.size/20+epsilon){
	    this.scale-=epsilon
	}
	//this.scale = this.target.size/20
	this.x = this.target.position.x - c.width/2*this.scale
	this.y = this.target.position.y - c.height/2*this.scale
    }
    set_target(target){
	this.target = target
    }
}

export var camera = new Camera(0, 0, 1.5, null)

File: fiddling/server/camera.py
from .user_inputs import GameKeys


class Camera:
    def __init__(self, x, y, scale, target, user_inputs):
        self.x = x
        self.y = y
        self.scale = scale
        self.target = target
        self.user_inputs = user_inputs
        self.size_x = 1
        self.size_y = 1

    def update(self, dt):
        if self.user_inputs.pressed_key(GameKeys.ZOOM_IN):
            self.scale *= 0.99
        elif self.user_inputs.pressed_key(GameKeys.ZOOM_OUT):
            self.scale *= 1.01

        self.size_x = self.target.position.size_x * self.scale
        self.size_y = self.target.position.size_y * self.scale
        self.x = self.target.position.x
        self.y = self.target.position.y
